### DISCUSSION TITLE
Spike Prime Hub Lautsprecher Lautstärke steuern

### DISCUSSION THREAD

**Discussion by msw-home** [pybricks/support#1794] [Q&A]
Kann man die Lautstärke von einem Spike Prime Hub mit der Blockprogrammierung steuern?
Wenn ja, wäre ein kurzes Beispiel hilfreich.
Gruß Manfred

**Comment by BertLindeman** [comment]
Ha Manfred,

I do not see a block to do that and tried to make a small python function to set the volume, but . . . 
I failed to do that.

Have this python function `hub_volume.py`:
```python

def set_hub_volume(hub, my_volume):
    hub.speaker.volume(my_volume)
```

I wanted to use it in an import block.
Problem was to get the hub into the function call.
Simple: the hub is in the variables...

So block program:
![afbeelding](https://github.com/user-attachments/assets/3bb34256-f498-4e15-b790-6c4ff4b798e5)

Hope this helps.

Bert

**Reply by msw-home** [reply]
![Screenshot 2024-08-30 194437](https://github.com/user-attachments/assets/ff8ff43f-f0d6-4c6c-9a0c-57bbbcad1f1a)
Wo ist mein (Denk)fehler?

**Reply by laurensvalk** [reply]
Bert is suggesting that you create a new Python file called `hub_volume` with the contents he provided.

Then the block program you made will execute that function, and thus decrease the volume.

Basically, this is a nice way to try functions that exist in Python but not (yet) as a block.

**Comment by BertLindeman** [comment]
Like this:

Screenshot of a part of pybricks:
![afbeelding](https://github.com/user-attachments/assets/d1e95f2b-beb1-476b-97ff-54b104803d4b)

**Comment by msw-home** [comment]
Danke für diese Problemlösung, ich kann kein Python, kann ich meinem Enkel auch nicht als Lösung anbieten.
Wir versuchen uns gerade in die Blockprogrammierung einzuarbeiten, alles andere überfordert mich (69 Jahre) und ihn (8 Jahre).
Er hat dann in 3-5 Jahren die besseren Chancen, Python zu verstehen und/oder darin zu programmieren.
Aber wie schon geschrieben, danke an alle, die sich unserer Probleme annehmen.
VG Tom und Opa
